contracts/actions/curveusd/helpers/CurveUsdHelper.sol

```diff
@@ -4,16 +4,31 @@ pragma solidity =0.8.10;
 import {ICrvUsdController, ILLAMMA, ICrvUsdControllerFactory, ICurveUsdSwapper} from "../../../interfaces/curveusd/ICurveUsd.sol";
 import "../../../interfaces/curve/IAddressProvider.sol";
 
+import "../../../interfaces/IERC20.sol";
+import "../../../DS/DSMath.sol";
 import "./MainnetCurveUsdAddresses.sol";
 import "../../../utils/TokenUtils.sol";
 
-contract CurveUsdHelper is MainnetCurveUsdAddresses {
+contract CurveUsdHelper is MainnetCurveUsdAddresses, DSMath {
     using TokenUtils for address;
 
     error CurveUsdInvalidController();
 
     bytes4 constant CURVE_SWAPPER_ID = bytes4(keccak256("CurveUsdSwapper"));
 
+    function getCollateralRatio(address _user, address _controllerAddr) public view returns (uint256 collRatio, bool isInSoftLiquidation) {
```

Comment:
Would slightly refactor this function to make it more readable. Something like this


        ICrvUsdController crvController = ICrvUsdController(_controllerAddr);
        
        uint256 debt = crvController.debt(_user);
        if (debt == 0) return (0, false);
        
        (uint256 crvUsdCollAmount, uint256 collAmount) = getCollAmountsFromAMM(_controllerAddr, _user);
        isInSoftLiquidation = crvUsdCollAmount > 0;
        
        address amm = crvController.amm();
        address collToken = crvController.collateral_token();
        uint256 oraclePrice = ILLAMMA(amm).price_oracle();
        uint256 assetDec = IERC20(collToken).decimals();
        
        uint256 collAmountWAD = assetDec > 18 ? (collAmount / 10 ** (assetDec - 18)) : (collAmount * 10 ** (18 - assetDec));
        collRatio = wdiv(wmul(collAmountWAD, oraclePrice) + crvUsdCollAmount, debt);
    
But this is totally opinionated